vault/barrier_aes_gcm.go

```diff
@@ -472,8 +474,8 @@ func (b *AESGCMBarrier) Rotate(ctx context.Context) (uint32, error) {
 
 // CreateUpgrade creates an upgrade path key to the given term from the previous term
 func (b *AESGCMBarrier) CreateUpgrade(ctx context.Context, term uint32) error {
-	b.l.RLock()
-	defer b.l.RUnlock()
+	b.l.Lock()
```

Comment:
I think this should still be a RLock.  I believe there is no update to what this lock is protecting.

Reply:
I agree, it just felt kinda right for it to be a write lock since it's a not-common path. But, I'll change it back.